src/js/map/controls.js

```diff
@@ -110,12 +112,20 @@ class Controls {
 
 
   toggleClearButton() {
-    this.map.on("layeradd", () => this.showClearButton());
+    this.map.on("layeradd", () => {
+      if (this.mapConfig.blockInteractiveLegend){
```

Comment:
I think it is not necessary to hide the control each time. Instead of line 115 to 122 I would just do
this.map.on("layeradd", () => {
      if (! this.mapConfig.blockInteractiveLegend){
        this.showClearButton()
      }
    });

Reply:
you are right, it is easier. It is changed